### ISSUE TITLE
Modulation Matrix clears Dest 1 & 2 when clicked

### ISSUE THREAD

**Issue by xard-dev** [TheWaveWarden/odin2#325]
**Describe the bug**
Modulation Matrix **Dest 1** and **Dest 2** slots reset current value when clicked with primary click.

**To Reproduce**
1. Open Odin2 UI and Modulation Matrix.
2. Assing values to Source, Dest 1, Dest 2 and Scale.
3. Try to change Source, Scale, Dest 1 and Dest 2

**Expected behavior**
Primary clicking assigned value a context menu is presented with selected option checked.

**Desktop (please complete the following information):**
- Ubuntu 20.04
- Ardour 6.3
- Odin 2.2.4
- Plugin Format LV2

**Additional information**
Secondary click clears the field values in Dest 1 and Dest 2 which probably is part of the bug but actually it could be useful feature in itself.

**Comment by TheWaveWarden** [comment]
Yes, like you correctly assesed, the menu is built fromt he ground regarding the currentl available modules. The underlying structure `Juce::PopupMenu` doesn't allow for the value to be stored.

Changing this would probably take some rethinking of the way these menus are reconstructed.

ATM I don't see an obvious solution.

**Comment by xard-dev** [comment]
Thanks for the information!

From the response I think this problem can be divided into two key areas:
1. Context menu having correct item checked.
2. Instrument data updates before an item from the menu has been selected.

The first one requires a stable and reliable way to carry the matching information over from the instrument data which can be quite tricky depending on how the system works (not very familiar with JUCE UI components).

However the second item is the major annoyance for the end user. Depending how the widget behaves it might be easier to fix.
